src/Diagnostics.DataProviders/KustoClient.cs

```diff
@@ -92,56 +79,49 @@ public async Task<DataTable> ExecuteQueryAsync(string query, string cluster, str
                     dataSet = JsonConvert.DeserializeObject<DataTableResponseObjectCollection>(responseContent);
                 }
             }
-            catch(Exception ex)
+            catch (Exception ex)
             {
-                kustoApiException = ex;
+                LogKustoQuery(query, cluster, operationName, timeTakenStopWatch, kustoClientId, ex, dataSet);
                 throw;
             }
-            finally
-            {
-                timeTakenStopWatch.Stop();
-                var status = kustoApiException == null ? "Success" : "Failed";
 
-                object stats = null;
-                if (dataSet != null && dataSet.Tables != null && dataSet.Tables.Count() >= 4)
-                {
-                    var statisticsTable = dataSet.Tables.ToArray()[dataSet.Tables.Count() - 2];
-                    if (statisticsTable.Rows.GetLength(0) >= 2 && statisticsTable.Rows.GetLength(1) >= 5)
-                    {
-                        stats = statisticsTable.Rows[1, 4];
-                    }
-                }
+            LogKustoQuery(query, cluster, operationName, timeTakenStopWatch, kustoClientId, null, dataSet);
 
-                DiagnosticsETWProvider.Instance.LogKustoQueryInformation(
-                   operationName ?? "None",
-                   _requestId,
-                   $"KustoQueryRequestId:{kustoClientId},Status:{status},TimeTaken:{timeTakenStopWatch.ElapsedMilliseconds},Cluster:{cluster}",
-                   timeTakenStopWatch.ElapsedMilliseconds,
-                   JsonConvert.SerializeObject(stats) ?? string.Empty,
-                   query,
-                   kustoApiException != null ? kustoApiException.GetType().ToString() : string.Empty,
-                   kustoApiException != null ? kustoApiException.ToString() : string.Empty);
-            }
+            return dataSet?.Tables == null ? new DataTable() : dataSet.Tables.FirstOrDefault().ToDataTable();
+        }
 
-            
+        private void LogKustoQuery(string query, string cluster, string operationName, Stopwatch timeTakenStopWatch, string kustoClientId, Exception kustoApiException, DataTableResponseObjectCollection dataSet)
+        {
+            timeTakenStopWatch.Stop();
```

Comment:
I suggest just keeping this inside of the finally clause and let the `LogKustoQuery` function just be focused on the single responsibility of logging.

Reply:
Definitely, that slipped under my radar when I used the VS `refactor to function` operation